code/script.js

```diff
@@ -1,51 +1,221 @@
+  const flavourConfirmation = (button) => {
+    button.addEventListener("click", () => {
+      userResponse(`I would love to have ${button.value}!`);
+      setTimeout(botResponse(`${button.value} is a really great choice!`), 1000);
+      moveToScoopQuestion();
+    });
+  }
+
+  // Invoke the functions with buttons
+  flavourConfirmation(chocolateButton);
+  flavourConfirmation(strawberryButton);
+  flavourConfirmation(vanillaButton);
+}
+
+//------------QUESTION 3-------------
+// Bot shows options for the number of scoops upon receiving user's choice of flavour
+const questionAboutNumberOfScoops = () => {
+  botResponse("How many scoops would you like to have?");
+
+  // Create drop-down list with options
+  inputWrapper.innerHTML += `
+    <select name="scoops" id="scoops">
+      <option value="select-one" selected disabled>Select one option...</option>
+      <option value="One">One</option>
+      <option value="Two">Two</option>
+      <option value="Three">Three</option>
+    </select>
+    `;
+  
+  // Store user's choice in a variable
+  const scoops = document.getElementById("scoops");
+
+  // Add event listener to change of the option in the drop-down list so user's choice is displayed and bot shows confirmation message
+  scoops.addEventListener("change", () => {
+    userResponse(`${scoops.value} scoop(s) please!`);
+    setTimeout(botResponse(`${scoops.value} scoop(s), got it!`), 1000);
+    
+    // Clear the dropdown list and move to next question
+    scoops.remove();
+    setTimeout(() => questionAboutToppings(), 1000);
+  });
 }
 
-// Set up your eventlisteners here
+//------------QUESTION 4-------------
+// Bot shows options for toppings upon receiving user's answer about number of scoops
+const questionAboutToppings = () => {
+  botResponse("What about toppings? Would you like to add some?");
+
+  // Create buttons for toppings
+  inputWrapper.innerHTML += `
+    <button id="sprinkles" value="Sprinkles">Sprinkles</button>
+    <button id="sauce" value="Chocolate sauce">Chocolate sauce</button>
+    <button id="fruit" value="Fresh fruit">Fresh fruit</button>
+    `;
+  
+  // Store the buttons in variables
+  const sprinklesButton = document.getElementById("sprinkles");
+  const sauceButton = document.getElementById("sauce");
+  const fruitButton = document.getElementById("fruit");
+
+  // Function to clear the buttons and move to the next question
+  const moveToLastQuestion = () => {
+    sprinklesButton.remove();
+    sauceButton.remove();
+    fruitButton.remove();
+    setTimeout(() => questionAboutConfirmation(), 1000);
+  };
+
+  // Function to add event listener to click on buttons so user's choice is displayed and bot shows confirmation message
+  const toppingConfirmation = (button) => {
+    button.addEventListener("click", () => {
+      userResponse(`${button.value} please. That's my favourite!`);
```

Comment:
here also you can store the value if you want to extend your question and use this value